corehq/apps/app_manager/views/formdesigner.py

```diff
@@ -98,6 +101,33 @@ def form_source_legacy(request, domain, app_id, module_id=None, form_id=None):
     return _get_form_designer_view(request, domain, app, module, form)
 
 
+def _get_form_submit_history_url(request, domain, app, module, form):
+    submission_history_view = 'corehq.apps.reports.standard.inspect.SubmitHistory'
+    if not has_permission_to_view_report(request.couch_user, domain, submission_history_view):
+        return None
+
+    end_date = datetime.now()
+    start_date = end_date - timedelta(days=30)
+
+    base_url = reverse('project_report_dispatcher', args=[domain, 'submit_history'])
+    params = {
+        'form_status': 'active',
+        'form_app_id': app.id,
+        'form_module': module.id,
+        'form_xmlns': form.xmlns,
+        'startdate': start_date.strftime('%Y-%m-%d'),
+        'enddate': end_date.strftime('%Y-%m-%d'),
+    }
+    return _make_url(base_url, params)
+
+
+def _make_url(base_url, params):
```

Comment:
I believe you can use `corehq.util.view_utils` `reverse` for this https://github.com/dimagi/commcare-hq/blob/e9cef132ecf35a20ccc3b5dcb955b31a5fbe46c5/corehq/util/view_utils.py#L119